src/OpenLoco/src/Input/Keyboard.cpp

```diff
@@ -168,11 +170,16 @@ namespace OpenLoco::Input
 
     void enqueueText(const char* text)
     {
-        if (text != nullptr && text[0] != '\0')
+        using namespace Localisation;
+
+        if (text == nullptr || text[0] == '\0')
         {
-            uint32_t index = _keyQueueLastWrite;
-            _keyQueue[index].charCode = text[0];
+            return;
         }
+
+        uint32_t index = _keyQueueLastWrite;
+        auto unsignedText = reinterpret_cast<const unsigned char*>(text);
```

Comment:
can you at least use uint8_t instead of unsigned char since that is what we use throughout the codebase.